mne_bids/read.py

```diff
@@ -166,3 +157,73 @@ def read_raw_bids(bids_fname, bids_root, verbose=True):
         raw.info['bads'] = list(unique_bads)
 
     return raw
+
+
+def get_head_mri_trans(bids_fname, bids_root):
+    """Produce transformation matrix from MEG and MRI landmark points.
+
+    Will attempt to read the landmarks of Nasion, LPA, and RPA from the sidecar
+    files of (i) the MEG and (ii) the T1 weighted MRI data. The two sets of
+    points will then be used to calculate a transformation matrix from HEAD
+    coordinates to MRI coordinates.
+
+    Parameters
+    ----------
+    bids_fname : str
+        Full name of the MEG data file (not a path)
+    bids_root : str
+        Path to root of the BIDS folder
+
+    Returns
+    -------
+    trans : instance of mne.transforms.Transform
+        The data transformation matrix from HEAD to MRI coordinates
+
+    """
+    try:  # pragma: no cover
+        import nibabel as nib
+    except ImportError:  # pragma: no cover
+        raise ImportError('This function requires nibabel.')
+
+    # Get the sidecar file for MRI landmarks
+    t1w_json_path = _find_matching_sidecar(bids_fname, bids_root, 'T1w.json')
+
+    # Get MRI landmarks from the JSON sidecar
+    with open(t1w_json_path, 'r') as f:
+        t1w_json = json.load(f)
+    mri_coords_dict = t1w_json['AnatomicalLandmarkCoordinates']
+    mri_landmarks = np.asarray((mri_coords_dict['LPA'],
+                                mri_coords_dict['NAS'],
+                                mri_coords_dict['RPA']))
+
+    # The MRI landmarks are in "voxels". We need to convert the to the
+    # neuromag RAS coordinate system in order to compare the with MEG landmarks
+    # see also: `mne_bids.write.write_anat`
+    t1w_path = t1w_json_path.replace('.json', '.nii')
+    if not op.exists(t1w_path):
+        t1w_path += '.gz'  # perhaps it is .nii.gz? ... else raise an error
+    if not op.exists(t1w_path):
+        raise RuntimeError('Could not find the T1 weighted MRI associated '
+                           'with "{}". Tried: "{}" but it does not exist.'
+                           .format(t1w_json_path, t1w_path))
```

Comment:
This could be a surprising error message since the user provided .nii but in the error message they see .nii.gz

Reply:
In this function the user does not provide any MRI file, they point to an MEG file name ... and to the BIDS root path.

Also, I am checking both, for `.nii` and `.nii.gz` if one of them works: no problem, if neither works ... there is no harm in telling them that we looked for `.nii.gz` and did not find it.